G1ANT.Addon.Net/Models/AttachmentModel.cs

```diff
@@ -0,0 +1,44 @@
+﻿using MailKit;
+using MimeKit;
+
+namespace G1ANT.Addon.Net.Models
+{
+    public class AttachmentModel : IAttachmentModel
+    {
+        public string Name { get; }
+        public long Size { get; }
+        public string Type { get; }
+        public MimeEntity MimeEntity { get; }
+
+        const string emptyAttachmentDefaultName = "unnamed-attachment";
+
+        public AttachmentModel(MimeEntity mimeEntity)
+        {
+            Name = GetAttachmentName(mimeEntity);
+            Size = GetAttachmentSize(mimeEntity);
+            Type = GetAttachmentMediaTypeName(mimeEntity);
+            MimeEntity = mimeEntity;
+        }
+
+        private string GetAttachmentName(MimeEntity mimeEntity)
```

Comment:
MimeEntity is already a property of this object.
This class is a model. It should have state and should use it. It should not have methods that perform actions on - possibly - external data, because this is how services/business facades work.
TL;DR; remove arguments of all methods except constructor.